BootloaderCorePkg/Stage1B/Stage1B.c

```diff
@@ -419,6 +478,14 @@ SecStartup2 (
   // Perform pre-config board init
   BoardInit (PreConfigInit);
 
+  // Check if recovery is needed, if not in recovery path already
+  GetStateMachine (&StateMachine);
+  DEBUG ((DEBUG_INFO, "Current FW update state machine: %d\n", StateMachine));
+  if (PcdGetBool (PcdSblResiliencyEnabled) && StateMachine != FW_UPDATE_SM_RECOVERY && !IsRecoveryTriggered ()) {
+    CheckForTcoTimerFailures ();
+    CheckForAcmFailures (StateMachine);
+  }
```

Comment:
since PcdSblResiliencyEnabled is not supported on some platform (e.g. APL), so all the core changes for this feature (including GetStateMachine()) should be under the case PcdSblResiliencyEnabled  is enabled.